### PR TITLE
Add a floor for jobqs and add docs

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: docs-md/development/eventer.md [hunk 1]
```diff
@@ -149,6 +149,21 @@ The return value of callbacks for timer events should always be 0.
 
 ### Asynchronous Events
 
+Asynchronous events within libmtev run within the context of an `eventer_jobq_t`.
+Programmatically, or via configuration, these job queues can be created.  The
+context of the job queue can control how many threads are running and how many
+events may be queued using an advisory limit.  Job queue concurrency can be
+controlled are run-time.
```
Comment (rileyberton) [inline docs-md/development/eventer.md]: s/are/at

File: docs-md/development/eventer.md [hunk 2]
```diff
@@ -149,6 +149,21 @@ The return value of callbacks for timer events should always be 0.
...
+
+Job queues have several attributes related to concurrency: min, floor, desired, and max.
+
+The min and the max describe "functional correctness" constraints on the jobq.  For example,
+if it would "malfunction" if more than one job ran concurrency, then the max should be set
```
Comment (rileyberton) [inline docs-md/development/eventer.md]: s/concurrency/concurrently

File: src/eventer/eventer_jobq.c [hunk 3]
```diff
@@ -875,12 +876,23 @@ void eventer_jobq_set_max_backlog(eventer_jobq_t *jobq, uint32_t max) {
   jobq->max_backlog = max;
 }
 
+void eventer_jobq_set_floor(eventer_jobq_t *jobq, uint32_t floor_concurrency) {
+  if(floor_concurrency > jobq->min_concurrency)
+  jobq->min_concurrency = floor_concurrency;
```
Comment (rileyberton) [inline src/eventer/eventer_jobq.c]: I don't see how min and floor are different.  Floor modifies min here.  And min modifies floor in set_min_max

Comment (postwait) [reply src/eventer/eventer_jobq.c]: floor <= min <= desired <= max


Floor is the minimum number of threads when idle.

Desired is the max number of threads when there is a job backlog.

Desired can only be changed within [min, max]

Comment (rileyberton) [reply src/eventer/eventer_jobq.c]: ok, that makes sense.

**Pull request merged**